code/iaas/simple-allocator/src/main/java/io/cattle/platform/simple/allocator/SimpleAllocator.java

```diff
@@ -84,6 +91,11 @@
     @Inject
     GenericMapDao mapDao;
 
+    private String cachedRequiredVersion;
+    private Integer requiredMajor;
+    private Integer requiredMinor;
+    private Integer requiredPatch;
```

Comment:
we probably shouldn't worry about patch version. Unless we are planning to re-use this code for other version checks (in this case, it would probably make sense to add it to some utils class)